kaldb/src/main/java/com/slack/kaldb/metadata/schema/SchemaUtil.java

```diff
@@ -0,0 +1,51 @@
+package com.slack.kaldb.metadata.schema;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import com.slack.kaldb.proto.schema.Schema;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.commons.text.StringSubstitutor;
+import org.apache.commons.text.lookup.StringLookup;
+
+public class SchemaUtil {
+
+  public static Schema.PreprocessorSchema parseSchema(Path schemaPath) throws IOException {
+    String filename = schemaPath.getFileName().toString();
+    if (filename.endsWith(".yaml")) {
+      return parseSchemaYaml(Files.readString(schemaPath), System::getenv);
+    } else if (filename.endsWith(".json")) {
```

Comment:
We can leave the json for now, but I think we'll be removing json support more broadly moving forward for our config files.